homeassistant/components/prometheus/__init__.py

```diff
@@ -276,6 +276,23 @@ def _handle_input_boolean(self, state):
         value = self.state_as_number(state)
         metric.labels(**self._labels(state)).set(value)
 
+    def _handle_input_number(self, state):
+        unit = self._unit_string(state.attributes.get(ATTR_UNIT_OF_MEASUREMENT))
+
+        metric = self._metric(
+            "input_number_state",
```

Comment:
Since input numbers can have different units of measurement, we'll need to have it as the metric name suffix: `f"input_number_state_{unit}"`

Reply:
Can't it have some special chars? Like `°C`

Reply:
This should not be a problem, I believe, since special chars get stripped in `_sanitize_metric_name`. Also, temperature units are handled separately in `_unit_string`.

Reply:
done